### PR TITLE
add CVE patch for XZ 5.2.5 + attempt to fix symbol patch for all OSs

### LINKED ISSUE #15846: Patches for XZ 5.2.5
There's a CVE patch for XZ (see https://tukaani.org/xz/).

It was also noted in #14991 that the patch we currently use for XZ is causing problems. I believe it should be changed to provide the symbols for `XZ_5.2` (perhaps as well as `XZ_5.2.2` to retain backwards compatibility?)

### PR DESCRIPTION
**Pull request by ocaisa**
Fixes #15846

(created using `eb --new-pr`)

### CODE REVIEW

**Comment by ocaisa** [comment]
This is (I believe) a safe patch as the 5.3 series is the current development version...but I have never seen a `.map` file before this PR!

**Comment by ocaisa** [comment]
Test report by @ocaisa
**SUCCESS**
Build succeeded for 5 out of 5 (5 easyconfigs in total)
login1 - Linux Rocky Linux 8.5, x86_64, Intel(R) Xeon(R) CPU E5-2690 v3 @ 2.60GHz, Python 3.6.8
See https://gist.github.com/4c4ec00e097b3c9a54f686ddf75b8be4 for a full test report.

File: easybuild/easyconfigs/x/XZ/XZ-5.2.5_compat-libs.patch [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
++	lzma_stream_encoder_mt_memusage;
++} XZ_5.0;
++
++XZ_5.2.2 {
+ global:
+ 	lzma_block_uncomp_encode;
+ 	lzma_cputhreads;
+ 	lzma_get_progress;
+-	lzma_stream_encoder_mt;
+-	lzma_stream_encoder_mt_memusage;
```
Comment (ocaisa) [inline easybuild/easyconfigs/x/XZ/XZ-5.2.5_compat-libs.patch:27]: The older patch did not remove these lines but then the linker throws some warnings about them already being declared in `XZ_5.1.2alpha`

File: easybuild/easyconfigs/x/XZ/XZ-5.2.5_compat-libs.patch [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+ 
+ local:
+ 	*;
+-} XZ_5.0;
++} XZ_5.1.2alpha;
++
++XZ_5.2 {
```
Comment (ocaisa) [inline easybuild/easyconfigs/x/XZ/XZ-5.2.5_compat-libs.patch:34]: I'm not sure if this is the right way to do this but hopefull the idea is clear

**Comment by ocaisa** [comment]
Confirmation in https://github.com/easybuilders/easybuild-easyconfigs/issues/14991#issuecomment-1184834493 that new patch still solves the old problem on CentOS7

**Comment by jfgrimm** [comment]
Test report by @jfgrimm
**SUCCESS**
Build succeeded for 13 out of 13 (6 easyconfigs in total)
node029.pri.viking.alces.network - Linux CentOS Linux 7.9.2009, x86_64, Intel(R) Xeon(R) Gold 6138 CPU @ 2.00GHz (skylake_avx512), Python 3.6.8
See https://gist.github.com/e67047f54d5ac159b0f84ed34caf71c1 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 6 out of 6 (6 easyconfigs in total)
node3521.doduo.os - Linux RHEL 8.4, x86_64, AMD EPYC 7552 48-Core Processor (zen2), Python 3.6.8
See https://gist.github.com/7c3776e5fe9b27299ff9aef4036984e1 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 6 out of 6 (6 easyconfigs in total)
node3703.slaking.os - Linux RHEL 8.4, x86_64, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz (haswell), Python 3.6.8
See https://gist.github.com/26d43c31cf2866d3a080dcbab535ef3a for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 2 out of 6 (6 easyconfigs in total)
node3703.slaking.os - Linux CentOS Linux 7.9.2009, x86_64, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz (haswell), Python 2.7.5
See https://gist.github.com/a1b94f818a5e124d981b79f7ebecb5f4 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 6 out of 6 (6 easyconfigs in total)
node3703.slaking.os - Linux CentOS Linux 7.9.2009, x86_64, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz (haswell), Python 2.7.5
See https://gist.github.com/e918f52f72af67a1e651a01a86cba46b for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 6 out of 6 (6 easyconfigs in total)
node3521.doduo.os - Linux Ubuntu 20.04.4 LTS (Focal Fossa), x86_64, AMD EPYC 7552 48-Core Processor (zen2), Python 3.8.10
See https://gist.github.com/ab8c7818c8fd5f62dfe272f0f3b666bc for a full test report.

**Review by boegel — APPROVED** [review]
Tested on RHEL8, CentOS 7 and Ubuntu 20.04, looks good!

**Comment by boegel** [comment]
Going in, thanks @ocaisa!

**Pull request merged**